src/osw_ui.cpp

```diff
@@ -298,20 +304,44 @@ void OswUI::OswUIProgress::draw() {
 size_t OswUI::OswUINotification::count{};
 
 OswUI::OswUINotification::OswUINotification(std::string message, bool isPersistent)
-    : message{message.c_str()}, endTime{isPersistent ? millis() + 300'000 : millis() + 5'000} {
-    id = count;
+    : id(count), message{message.c_str()}, lines(countLines(message)), endTime{millis() + (isPersistent ? 300'000 : 5'000) * lines} {
     ++count;
 }
 
+/**
+ * @brief This function is used in constructor to calculate the constant value
+ *
+ * @param message
+ * @return unsigned char
+ */
+unsigned char OswUI::OswUINotification::countLines(const std::string& message) const {
+    unsigned char lines = 1;
+    for (auto c: message) {
+        if (c == '\n') {
+            ++lines;
+        }
+    }
+    return lines;
+}
+
+/**
+ * @brief Get the draw height based on line count with additional padding
+ *
+ * @return unsigned char
+ */
+unsigned char OswUI::OswUINotification::getDrawHeight() const {
+    return 4 + 8 * lines + 4; // padding, line height, padding
+}
+
 void OswUI::OswUINotification::draw(unsigned y) const {
-    // TODO
-    // * handle too long texts by adding a scroll animation?
-    // * newlines may change a notifications height, so the y position should be calculated based on the current height of the notification -> return that instead of void
+    // TODO handle too long texts by adding a scroll animation?
     Graphics2DPrint* gfx = OswHal::getInstance()->gfx();
-    gfx->fillFrame(0, y, DISP_W, this->sDrawHeight, OswUI::getInstance()->getBackgroundColor());
+    auto height = this->getDrawHeight();
+    gfx->fillFrame(0, y, DISP_W, height, OswUI::getInstance()->getBackgroundColor());
```

Comment:
same, rename `y`.

Reply:
Same issue here... `yStart`? `posY`? Idk, take a look into the GFX2D library, they also use one-letter names for coordinate values there.